code/modules/components/ai/hostile/inject_poison.dm

```diff
@@ -0,0 +1,23 @@
+/datum/component/ai/inject_poison
+	var/poison_type = "" // STOXIN, etc
+	var/poison_per_bite = 0 // Mols to inject
+	var/inject_prob = 0 // Chance to inject, -1 = ALWAYS
+	var/max_poison = 0 // Maximum mols in target's blood. 0 = INF
+
+/datum/component/ai/inject_poison/RecieveSignal(var/message_type, var/list/args)
+	switch(message_type)
+		if("attacking target") // list("target"=A)
```

Comment:
Could probably do it with a precompile step, but that starts getting into "clusterfuck of a buildsystem" territory.